File: 0575-distribute-candies/0575-distribute-candies.java
class Solution {
    public int distributeCandies(int[] candyType) {
        int n = candyType.length;
        int maxTypes = n/2;
        Set<Integer> uniqueCandies = new HashSet<>();
        for(int candy: candyType){
            uniqueCandies.add(candy);
        }
        return Math.min(maxTypes, uniqueCandies.size());
    }
}

File: 0575-distribute-candies/0575-distribute-candies.cpp
class Solution {
public:
    int distributeCandies(vector<int>& candyType) {
        unordered_set<int> uniqueTypes;
        int n = candyType.size();
        int maxTypes = n/2;
        for(int candy: candyType){
            uniqueTypes.insert(candy);
        }
        return min(maxTypes, (int)uniqueTypes.size());
    }
};